conf4j-core/src/main/java/com/sabre/oss/conf4j/converter/standard/LocalDateTimeTypeConverter.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * MIT License
+ *
+ * Copyright 2017 Sabre GLBL Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sabre.oss.conf4j.converter.standard;
+
+import com.sabre.oss.conf4j.converter.TypeConverter;
+
+import java.lang.reflect.Type;
+import java.time.DateTimeException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Map;
+
+import static java.time.LocalDateTime.parse;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This class converts {@link LocalDateTime} to/from string.
+ */
+public class LocalDateTimeTypeConverter implements TypeConverter<LocalDateTime> {
+
+    @Override
+    public boolean isApplicable(Type type, Map<String, String> attributes) {
+        requireNonNull(type, "type cannot be null");
+
+        return type instanceof Class<?> && LocalDateTime.class.isAssignableFrom((Class<?>) type);
+    }
+
+    @Override
+    public LocalDateTime fromString(Type type, String value, Map<String, String> attributes) {
+        requireNonNull(type, "type cannot be null");
+
+        if (value == null) {
+            return null;
+        } else {
+            try {
+                return attributes.containsKey("format")
```

Comment:
I'm not sure if the conversion performance will be good enough - creating formatter for every invocation of the converter may be inefficient. I would recommend creating some kind of cache by format and reuse formatter between calls.

It seams to be a good idea to distinguish between errors caused by bad _format_ pattern format and value format and include in the error message the _format_ value.

The same comments apply to _toString() method too.

Reply:
When it comes to caching formatters I have few options:
- Upgrade commons-lang3 to version 3.6 and use provided Memoizer. Although it is very basic and it doesn't allow to limit numbers of entries.
- Use Cache from spring-context. It requires adding ~6MB dependency to module.
- Add external dependency like Guava. It will be used only in this single place for now.
- Use ConcurrentHashMap as simple cache.
Currently I am leaning towards last solution. Formats for date/time are rather limited. What is more, probably, most of the time user will be using few different formats.

What do you think? Should I go with the most lightweight solution (ConcurrentHashMap) or choose more complex one?